examples/maps/main.rs

```diff
@@ -0,0 +1,364 @@
+                        let buffer = SharedPixelBuffer::<Rgba8Pixel>::clone_from_slice(
+                            image.as_raw(),
+                            image.width(),
+                            image.height(),
+                        );
+                        slint::Image::from_rgba8(buffer)
+                    })
+                });
+            }
+        }
+    }
+
+    fn poll(&mut self, context: &mut Context, changed: &mut bool) {
+        self.loading_tiles.retain(|coord, future| {
+            let image = future.as_mut().poll(context);
+            match image {
+                Poll::Ready(image) => {
+                    self.loaded_tiles.insert(*coord, image);
+                    *changed = true;
+                    false
+                }
+                Poll::Pending => true,
+            }
+        })
+    }
+}
+
+struct State {
+    world: RefCell<World>,
+    main_ui: MainUI,
+    poll_handle: RefCell<Option<slint::JoinHandle<()>>>,
+}
+
+impl State {
+    fn do_poll(self: Rc<Self>) {
+        if let Some(handle) = self.poll_handle.take() {
+            handle.abort();
+        }
+        self.refresh_model();
+        slint::spawn_local(async move {
+            std::future::poll_fn(|context| {
+                let mut changed = false;
+                self.world.borrow_mut().poll(context, &mut changed);
+                if changed {
+                    self.refresh_model();
+                }
+                if self.world.borrow().loading_tiles.is_empty() {
+                    Poll::Ready(())
+                } else {
+                    Poll::Pending
+                }
+            })
+            .await;
+        })
+        .unwrap();
+    }
+
+    fn refresh_model(&self) {
+        let vec = VecModel::from(
+            self.world
+                .borrow()
+                .loaded_tiles
+                .iter()
+                .map(|(coord, image)| Tile {
+                    tile: image.clone(),
+                    x: (coord.x * TILE_SIZE) as f32,
+                    y: (coord.y * TILE_SIZE) as f32,
+                })
+                .collect::<Vec<Tile>>(),
+        );
+        self.main_ui.set_tiles(slint::ModelRc::new(vec));
+    }
+
+    fn set_viewport_size(&self) {
+        let world = self.world.borrow();
+        let zoom = world.zoom_level;
+        self.main_ui.set_zoom(zoom as _);
+        let world_size = (TILE_SIZE * (1 << zoom)) as f32;
+        self.main_ui.invoke_set_viewport(
+            -world.offset_x as f32,
+            -world.offset_y as f32,
+            world_size,
+            world_size,
+        );
+    }
+}
+
+fn main() {
+    let rt = tokio::runtime::Runtime::new().unwrap();
+    let _tokio = rt.enter();
+
+    let state = Rc::new(State {
+        world: RefCell::new(World::new()),
+        main_ui: MainUI::new().unwrap(),
+        poll_handle: None.into(),
+    });
+
+    {
+        let size = state.main_ui.window().size();
```

Comment:
@tronical  What size do you get there? If that's not correct, that might be the reason why it doesn't load the map at start.
(We'd need a way to detect geometry change ideally)

Reply:
The size is zero :-(

```
PhysicalSize {
    width: 0,
    height: 0,
}
```